src/client/v2/algod/setSyncRound.ts

```diff
@@ -1,22 +1,27 @@
 import JSONRequest from '../jsonrequest.js';
-import { HTTPClient } from '../../client.js';
 import IntDecoding from '../../../types/intDecoding.js';
+import { HTTPClient } from '../../client.js';
 
 export default class SetSyncRound extends JSONRequest {
   constructor(
     c: HTTPClient,
-    intDecoding: IntDecoding,
     private round: number
   ) {
-    super(c, intDecoding);
+    super(c);
   }
 
   path() {
     return `/v2/ledger/sync/${this.round}`;
   }
 
   async do(headers = {}) {
-    const res = await this.c.post(this.path(), null, undefined, headers);
+    const res = await this.c.post({
```

Comment:
Similar to the `KmdClient.post` suggestion but deeper down, should we change `HTTPClient` to make these call sites simpler, since they are the common case.  Seems like `parseBody` and `jsonOptions` should default to these, so you can go back to simple call sites for most api calls (or add a new method like `apiPost`)

Reply:
I'm also unhappy with the `HTTPClient` methods at the moment, though I didn't know whether I should address them in this PR.

My inclination is to simplify those methods so that they just return the raw `Uint8Array` bytes, and perhaps introduce new methods which would return parsed JSON.

So it would be something like `HTTPClient.get` and `HTTPClient.getJSON` -- still thinking through this